netcore/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs

```diff
@@ -112,11 +112,17 @@ internal static bool IsBitwiseEquatable<T> ()
 		}
 
 		[Intrinsic]
-		internal static unsafe bool ObjectHasComponentSize (object obj)
+		internal static bool ObjectHasComponentSize (object obj)
 		{
 			throw new NotImplementedException ();
 		}
 
+		[Intrinsic]
```

Comment:
Tested both intrinsic and its fallback on arrays of different types - works as expected.